backend/scripts/chat_feedback_dump.py

```diff
@@ -0,0 +1,239 @@
+#     user_email: str
+#     name: str | None
+#     messages: list[MessageSnapshot]
+#     persona_name: str | None
+#     time_created: datetime
+#     flow_type: SessionType
+
+
+def create_new_chat_session(danswer_url: str, api_key: str | None) -> int:
+    headers = {"Authorization": f"Bearer {api_key}"} if api_key else None
+    session_endpoint = danswer_url + "/api/chat/create-chat-session"
+
+    response = requests.get(session_endpoint, headers=headers)
+    response.raise_for_status()
+
+    new_session_id = response.json()["chat_session_id"]
+    return new_session_id
+
+
+def manage_users(danswer_url: str, headers: dict[str, str] | None) -> AllUsersResponse:
+    endpoint = danswer_url + "/manage/users"
+
+    response = requests.get(
+        endpoint,
+        headers=headers,
+    )
+    response.raise_for_status()
+
+    all_users = AllUsersResponse(**response.json())
+    return all_users
+
+
+def get_chat_sessions(
+    danswer_url: str, headers: dict[str, str] | None, user_id: UUID
+) -> ChatSessionsResponse:
+    endpoint = danswer_url + "/admin/chat-sessions"
+
+    params: dict[str, Any] = {"user_id": user_id}
+    response = requests.get(
+        endpoint,
+        params=params,
+        headers=headers,
+    )
+    response.raise_for_status()
+
+    sessions = ChatSessionsResponse(**response.json())
+    return sessions
+
+
+def get_session_history(
+    danswer_url: str, headers: dict[str, str] | None, session_id: UUID
+) -> ChatSessionSnapshot:
+    endpoint = danswer_url + f"/admin/chat-session-history/{session_id}"
+
+    response = requests.get(
+        endpoint,
+        headers=headers,
+    )
+    response.raise_for_status()
+
+    sessions = ChatSessionSnapshot(**response.json())
+    return sessions
+
+
+def process_all_chat_feedback(danswer_url: str, api_key: str | None) -> None:
+    headers = {"Authorization": f"Bearer {api_key}"} if api_key else None
+
+    all_users = manage_users(danswer_url, headers)
+    if not all_users:
+        raise RuntimeError("manage_users returned None")
+
+    logger.info(f"Accepted users: {len(all_users.accepted)}")
+
+    user_ids: list[UUID] = [user.id for user in all_users.accepted]
+
+    for user_id in user_ids:
+        r_sessions = get_chat_sessions(danswer_url, headers, user_id)
+        logger.info(f"user={user_id} num_sessions={len(r_sessions.sessions)}")
+        for session in r_sessions.sessions:
+            try:
+                s = get_session_history(danswer_url, headers, session.id)
+            except requests.exceptions.HTTPError:
+                logger.exception("get_session_history failed.")
+
+            for m in s.messages:
+                logger.info(
+                    f"user={user_id} "
+                    f"session={session.id} "
+                    f"message={m.message} "
+                    f"feedback_type={m.feedback_type} "
+                    f"feedback_text={m.feedback_text}"
+                )
+
+
+if __name__ == "__main__":
```

Comment:
Might be worth adding a tiny comment to poke end user of the script in the right direction (something like "would need to retrieve the fastapiusersauth access token" and set that in the headers)